src/software/ai/passing/evaluation.cpp

```diff
@@ -58,9 +59,24 @@ double Passing::ratePass(const World& world, const Passing::Pass& pass,
     double pass_speed_quality = sigmoid(pass.speed(), min_pass_speed, 0.2) *
                                 (1 - sigmoid(pass.speed(), max_pass_speed, 0.2));
 
-    double pass_quality = static_pass_quality * friendly_pass_rating * enemy_pass_rating *
-                          shoot_pass_rating * in_region_quality *
-                          pass_time_offset_quality * pass_speed_quality;
+    // We ignore shoot pass quality if we're rating for a "RECEIVE_AND_DRIBBLE" pass
+    double pass_quality = 0;
+    switch (pass_type)
+    {
+        case RECEIVE_AND_DRIBBLE:
+            pass_quality = static_pass_quality * friendly_pass_rating *
+                           enemy_pass_rating * in_region_quality *
+                           pass_time_offset_quality * pass_speed_quality;
+            break;
+        case ONE_TOUCH_SHOT:
+            pass_quality = static_pass_quality * friendly_pass_rating *
+                           enemy_pass_rating * shoot_pass_rating * in_region_quality *
+                           pass_time_offset_quality * pass_speed_quality;
+            break;
+        default:
+            LOG(FATAL) << "Unhandled pass type given to `ratePass`";
```

Comment:
I know I wrote this........... but could we instead throw a `std::invalid_argument` here so that if this were to happen during a game we can catch the exception and carry on?